### PR TITLE
Guice: Multibinding task

### PR DESCRIPTION
**Pull request by kristiyanpetkov**
In my PR i use Guice Multiset module to bind an interface to more than one implementations.
Demo:
screenshot from 2016-07-13 17-53-19

MainModule:
screenshot from 2016-07-13 17-53-10

Console Output:
screenshot from 2016-07-11 10-16-25

### CODE REVIEW

File: src/main/java/multibinding/LeoMessi.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package multibinding;
+
+/**
+ * Created by Kristiyan Petkov  <kristiqn.l.petkov@gmail.com> on 07.07.16.
+ */
+public class LeoMessi implements Player {
+  public void play(Sport sport) {
+    if (sport.getType().equals("football")) {
```
Comment (mlesikov) [inline src/main/java/multibinding/LeoMessi.java]: In the real life, this situation should show you that the design is not fine, because you have to write if else in every implementation. Typically this can be avoid by using Command Pattern.
This note is not related with your current usage of the  guice multibinder .

File: src/main/java/multibinding/PlayerModule.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import com.google.inject.multibindings.Multibinder;
+
+/**
+ * Created by Kristiyan Petkov  <kristiqn.l.petkov@gmail.com> on 07.07.16.
+ */
+public class PlayerModule extends AbstractModule {
+  @Override
+  protected void configure() {
+    Multibinder<Player> binder = Multibinder.newSetBinder(binder(), Player.class);
+    binder.addBinding().to(MichaelJordan.class);
```
Comment (mlesikov) [inline src/main/java/multibinding/PlayerModule.java]: Another advantage of the multibinder is that the bindings can be `add` in a separate modules. So different implementations of a single interface can separate by their belonging module.
For example you can have users, that can be manipulated from your app.
When a new user is registered, then the app should manage to execute several actions:
 - send email address confirmation message;
 - save the creation details(creation date, created by,....)
 - creation default bank account
 ...

all that actions are different and if you call service for each of them, you code will become with many depandencies, hard for reading, hard for testing, hard to extend(add more actions)...

You can solve this using a simple interfece UserChagneListener. You can have 3(or more different dependencies) like UserLogger, UserEmailChecker, UserDefaultAccontCreator (maybe the names are not the best, but they shows what I mean). The interface can look like this:
```java 
interface   UserChagneListener { 
   void onUserCreation(User user)
}
```
then each implementation can be bind in different module and you just have to inject set  with all the binded implementations and call the correct method. Adding another implementation(another action)  becomes very easy, because you will add another implementation that will be bind in it's module. If you change one module wit another, then, removing the module, form the injector, directly means that the logic in will not be executed any more.

**Comment by mlesikov** [comment]
LGTM

**Comment by kristiyanpetkov** [comment]
@mlesikov , i made another variant of the task. please take a look when you have time.

File: src/main/java/multibinding/PlayerModule.java (hunk 2 above)
Comment (kristiyanpetkov) [inline src/main/java/multibinding/PlayerModule.java]: done.

File: src/main/java/multibinding/LeoMessi.java (hunk 1 above)
Comment (kristiyanpetkov) [inline src/main/java/multibinding/LeoMessi.java]: okay. i will have it in mind.

**Comment by panayotkulchev** [comment]
I see you added main module and installed the other 3 in it. 

LGTM!
